statsmodels/tsa/statespace/regime_switching/kim_filter.py

```diff
@@ -0,0 +1,635 @@
+import numpy as np
+from scipy.misc import logsumexp
+from scipy.optimize import linprog
```

Comment:
linprog is a 0.15.0 introduction, and since we don't really need it to compute the initial probabilities we should probably avoid it to maintain as much backwards compatibility as possible.